sound/soc/sof/intel/hda-dsp.c

```diff
@@ -314,6 +315,8 @@ static int hda_suspend(struct snd_sof_dev *sdev, int state)
 
 	/* disable hda bus irq and i/o */
 	snd_hdac_bus_stop_chip(bus);
+
+	snd_hda_set_power_save(hbus, SND_SOF_SUSPEND_DELAY_MS);
```

Comment:
we shouldn't muck with runtime_pm within a runtime_pm routine, this should be done on probe as done in Ranjani's patch.
Also this re-enabled runtime_pm in the suspend case and you disable it in resume, so this is very very odd.

Reply:
@plbossart This is trying to enforce more ordering by preventing codec to be runtime-suspended while controller is running. But for many reasons, this patch is a no-go.